### PR TITLE
docfx: remove `bottle :unneeded`

### PR DESCRIPTION
**Pull request by Amar1729**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See #75943.

### CODE REVIEW

**Comment by Amar1729** [comment]
Looks like no bottle for mono on arm. 
```
Error: mono: no bottle available!
You can try to install from source with:
  brew install --build-from-source mono
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```

File: Formula/docfx.rb [hunk 1]
```diff
@@ -5,8 +5,6 @@ class Docfx < Formula
   sha256 "95f96be280998e7ea6581d418c9b0d99dffb0c86bcca7f1dd257062163e0a79c"
   license "MIT"
 
-  bottle :unneeded
-
   depends_on "mono"
```
Comment (nandahkrishna) [inline Formula/docfx.rb:8]:
```suggestion
  depends_on arch: :x86_64
  depends_on "mono"
```

Comment (nandahkrishna) [reply Formula/docfx.rb:8]: This should restrict it to Intel Macs.

**Comment by Amar1729** [comment]
```
> sha256sum *.tar.gz

e1d454227a7f161c22bac66a7e29923dd20aaebe3492f84fa44be951683a3e81  docfx--2.56.7.big_sur.bottle.tar.gz
e1d454227a7f161c22bac66a7e29923dd20aaebe3492f84fa44be951683a3e81  docfx--2.56.7.catalina.bottle.tar.gz
e1d454227a7f161c22bac66a7e29923dd20aaebe3492f84fa44be951683a3e81  docfx--2.56.7.mojave.bottle.tar.gz
```

**Review by nandahkrishna — APPROVED** [review]

**Pull request closed without merging**